tests/frameworks/include/TestFramework.h

```diff
@@ -134,27 +206,47 @@
-                                for (const auto& item : values2) {                                                  \
-                                    combined.push_back(item);                                                       \
-                                }                                                                                   \
-                                                                                                                    \
-                                return ::testing::ValuesIn(combined);                                               \
-                                                                                                                    \
-                            }())
+#define WIN32_DISABLED_INSTANTIATE_TEST_CASE_P(instantiationName, testClass, values, disabledValues) \
+    INSTANTIATE_TEST_CASE_P(instantiationName, testClass, []() {                                     \
+        ::testing::internal::ParamGenerator<testClass::ParamType> values1 = values;                  \
+        ::testing::internal::ParamGenerator<testClass::ParamType> values2 = disabledValues;          \
+        std::vector<testClass::ParamType> combined;                                                  \
+        for (const auto& item : values1) {                                                           \
+            combined.push_back(item);                                                                \
+        }                                                                                            \
+                                                                                                     \
+        for (const auto& item : values2) {                                                           \
+            combined.push_back(item);                                                                \
+        }                                                                                            \
+                                                                                                     \
+        return ::testing::ValuesIn(combined);                                                        \
+                                                                                                     \
+    }())
 #endif
 
 // OSX defines for compatibility
 #ifndef boolean
 #define boolean Boolean
+#endif
+
+__if_not_exists(GetCurrentTestDirectory) {
+    static std::string GetCurrentTestDirectory() {
+        std::string tempBuffer;
+        uint32_t maxPath = _MAX_PATH;
+        tempBuffer.resize(maxPath);
+        _NSGetExecutablePath(&tempBuffer[0], &maxPath);
+        return tempBuffer.substr(0, tempBuffer.find_last_of('/')).c_str();
+    }
+}
+
+__if_not_exists(GetTestFullName) {
+    static std::string GetTestFullName() {
+        std::string toReturn;
+        return toReturn + ::testing::UnitTest::GetInstance()->current_test_info()->test_case_name() + "." +
```

Comment:
Why add `toReturn+` here? It looks like it's being done for `std::string` coercion? That's not great.